### PR TITLE
Fix issue #1401.

### PR DESCRIPTION
**Pull request by jcollins-g**
There's no framework currently for end-to-end testing of our template output outside manual review of the outputted HTML.   To make that easier:

Before:
before

After:
after

Regression for this issue is prevented by an assert for now.

### CODE REVIEW

File: lib/src/model.dart [hunk 1]
```diff
@@ -1040,6 +1040,14 @@ class Field extends ModelElement
     assert(enclosingElement != definingEnclosingElement);
   }
 
+  @override
+  String get documentation {
+    // Verify that we will show exactly one of the summaries.
+    Set<bool> assertCheck = new Set()..addAll([hasSetter, hasGetterNoSetter]);
```
Comment (devoncarew) [inline lib/src/model.dart]: I'm having trouble parsing this - what are we asserting here?

Comment (devoncarew) [inline lib/src/model.dart]: `Verify that hasSetter and hasGetterNoSetter are mutuablly exclusive.`?

**Comment by devoncarew** [comment]
lgtm

File: lib/src/model.dart (hunk 1 above)
Comment (jcollins-g) [inline lib/src/model.dart]: Done.

**Pull request merged**